Realm/Realm/DatabaseTypes/Geospatial/GeoPolygon.cs

```diff
@@ -0,0 +1,159 @@
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="GeoPolygon"/> class from a collection of <see cref="GeoPoint">GeoPoints</see>
+        /// with no holes.
+        /// </summary>
+        /// <param name="outerRing">The points representing the outer ring of the polygon.</param>
+        /// <remarks>
+        /// <paramref name="outerRing"/> must contain at least 3 unique points. The first and the last point may be identical, but
+        /// no other duplicates are allowed. Each subsequent pair of points represents an edge in the polygon with the first and the
+        /// last points being implicitly connected.
+        /// </remarks>
+        public GeoPolygon(params GeoPoint[] outerRing)
+            : this(new[] { outerRing })
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="GeoPolygon"/> class with an outer ring and a collection of holes.
+        /// </summary>
+        /// <param name="outerRing">
+        /// A collection of <see cref="GeoPoint">GeoPoints</see> representing the outer ring of the polygon.
+        /// </param>
+        /// <param name="holes">
+        /// A collection of collections of <see cref="GeoPoint">GeoPoints</see> representing the inner rings of the polygon.
+        /// </param>
+        /// <remarks>
+        /// <paramref name="outerRing"/> must contain at least 3 unique points. The first and the last point may be identical, but
+        /// no other duplicates are allowed. Each subsequent pair of points represents an edge in the polygon with the first and the
+        /// last points being implicitly connected.
+        /// <br/>
+        /// Each collection in <paramref name="holes"/> must contain at least 3 unique points with the same rules as for <paramref name="outerRing"/>.
+        /// <br/>
+        /// No two rings may intersect or share an edge, though they may share vertices.
+        /// <br/>
+        /// A point is considered "inside" the polygon if it is contained by an odd number of rings and "outside" if it's contained
+        /// by an even number of rings.
+        /// </remarks>
+        public GeoPolygon(IEnumerable<GeoPoint> outerRing, params IEnumerable<GeoPoint>[] holes)
+            : this(new[] { outerRing.ToList().AsReadOnly() }.Concat(holes.Select(h => h.ToList().AsReadOnly())).ToArray())
+        {
+        }
+
+        private GeoPolygon(IReadOnlyList<GeoPoint>[] linearRings)
+        {
+            foreach (var polygon in linearRings)
+            {
+                Argument.Ensure(polygon.Count > 3, $"Each linear ring (both the outer one and any holes) must have at least 4 points, but {LinearRingToString(polygon)} only had {polygon.Count}.", nameof(linearRings));
+                Argument.Ensure(polygon[0] == polygon[polygon.Count - 1], $"The first and the last points of the polygon {LinearRingToString(polygon)} must be the same.", nameof(linearRings));
```

Comment:
Same here, I thought in this case the polygon would be considered close anyways

Reply:
That was the case at some point, but Core changed it, so we now need to explicitly close the polygon. Will update the docs.

Reply:
Actually, I can see that MongoDB is more relaxed than GeoJSON, so might be worth doing it for the SDKs as well.